apps/patient/src/views/Pharmacy.tsx

```diff
@@ -113,127 +279,152 @@ export const Pharmacy = () => {
-      lat = geo.lat;
-      lng = geo.lng;
-
-      // Save location data for show more pharmacies
-      setLocation(loc);
-      setLatitude(lat);
-      setLongitude(lng);
+    const { address, lat, lng } = locationData;
+    if (!address || !lat || !lng) {
+      setLoadingPharmacies(false);
+      return;
     }
 
-    // Get pharmacies from our internal list
-    const locationData = {
-      latitude: lat,
-      longitude: lng,
-      radius: 25
-    };
-    const limit = 3;
-    const offset = pharmacyOptions.length;
+    setLocation(address);
+    setLatitude(lat);
+    setLongitude(lng);
+
+    // Get pharmacies from photon db
+    const pharmaciesResult = await getPharmacies(
+      {
+        latitude: lat,
+        longitude: lng,
+        radius: 25
+      },
+      30, // Set high to ensure indie's are found. Request time increase is minimal.
+      0,
+      token
+    );
+    if (pharmaciesResult === 'No pharmacies found near location') {
+      setLoadingPharmacies(false);
+      toast({
+        title: pharmaciesResult,
+        position: 'top',
+        status: 'warning',
+        duration: 5000,
+        isClosable: true
+      });
+      return;
+    }
+    if (!pharmaciesResult) {
+      setLoadingPharmacies(false);
+      return;
+    }
 
-    graphQLClient.setHeader('x-photon-auth', token);
+    // Save in case we fetched more than we initially showed
+    setFetchedPharmacies(pharmaciesResult);
 
-    let pharmaciesResults: any;
-    try {
-      pharmaciesResults = await graphQLClient.request(GET_PHARMACIES, {
-        location: locationData,
-        limit,
-        offset
-      });
-    } catch (error) {
-      console.error(JSON.stringify(error, undefined, 2));
-      console.log(error);
-      if (error?.response.errors[0].message === 'No pharmacies found near location') {
-        setShowingAllPharmacies(true);
+    // Prepare pharmacies to show
+    let newPharmacies: PharmacyType[] = pharmaciesResult;
+
+    // Feature pharmacies
+    newPharmacies = sortIndiePharmaciesFirst(
+      newPharmacies,
+      FEATURE_INDIES_WITHIN_RADIUS,
+      FEATURED_PHARMACIES_LIMIT
+    );
+
+    // Only enrich the first 3 to save time
+    const enrichedPharmacies: PharmacyType[] = await Promise.all(
+      newPharmacies.slice(0, 3).map(enrichPharmacy)
+    );
+    setPharmacyOptions(enrichedPharmacies);
+
+    setLoadingPharmacies(false);
+  };
+
+  const handleShowMore = async () => {
+    setLoadingPharmacies(true);
+
+    const updatedOptions = [...pharmacyOptions];
+
+    let totalNewEnriched = 0;
+
+    // Enrich existing pharmacies if we already have them
+    if (pharmacyOptions.length < fetchedPharmacies.length) {
+      const startIndex = pharmacyOptions.length;
+      const endIndex =
+        fetchedPharmacies.length - 1 < startIndex + 3
+          ? fetchedPharmacies.length - 1
+          : startIndex + 3;
+      for (let i = startIndex; i < endIndex; i++) {
+        if (totalNewEnriched === 3) break;
```

Comment:
if we show say 10, we're only showing rating/hours for 3 then?

Reply:
no, we only enrich 3 at a time so it doesn't take a long time to load